Source/SvgElement.cs

```diff
@@ -330,8 +328,9 @@ protected internal virtual bool PushTransforms(ISvgRenderer renderer)
 
             using (var transformMatrix = transforms.GetMatrix())
             {
-                if (_zeroMatrix.Equals(transformMatrix))
-                    return false;
+                using (var zeroMatrix = new Matrix(0f, 0f, 0f, 0f, 0f, 0f))
+                    if (zeroMatrix.Equals(transformMatrix))
+                        return false;

```

Comment:
Wouldn't it be better to leave the static zero matrix and create it lazily on first use? 
@gvheertum - I think that would not affect the gdiplus check, correct?

Reply:
It should indeed not affect the GDI+ handling if you do a lazy load (in the constructor or the property), since that would basically be the same as it is created now, being the Matrix getting constructed on the construction of the class or when called instead of when the runtime decides to create the static fields.

However, I think that it might limit readability and stability (in terms of future changes) if we use a static with a onetime initializer or a construction to get the field initially created in the constructor. Since the field resides in the class people can still accidentally use the non-wrapped field (thus ignoring a null check and initialization) or for example remove the initializer code or move the initializer back to the property/field (unless it's very well documented people tend to do this kind of small changes).

From a code quality perspective I think that having the matrix used this way is fine for now, because it's only used in one location/check, having additional code to initialize a constant that is only used once in a check might be a bit confusing and a bit over-done. It's nicer to do it in a static or const to limit allocation and memory footprint, but this might also be a bit over-engineering or micro-optimization (I don't have a clue of the memory impact of the Matrix, but I cannot really imagine it making a huge impact).

If I were to have this kind of constants in my code I would eventually move them to a specialized class/holder for the constant values (e.g. a SvgConstants class). This class can be a static class, and will expose static properties that do the initialization on the first request (if required together a lock { } mechanism if concurrency might be an issue). This keeps constants grouped and allows you to use constants throughout your library.

Reply:
Ok, thank you for the very thorough response - this answers my question! I agree that the current way is more readable and secure, and that we don't want premature optimization.

Makes me think about performance tests that could make sense... I noticed that in some of the example images rendering is very slow, so we certainly have some performance problems, though I have no idea what they are.